ibis/expr/api.py

```diff
@@ -4193,27 +4193,60 @@ def mutate(table, exprs=None, **mutations):
         for name, expr in sorted(mutations.items(), key=operator.itemgetter(0))
     )
 
+    selection_exprs_by_name = collections.OrderedDict()
```

Comment:
prob can just use a regular dict here (we are 3.7 and above)

Reply:
Makes sense, fixed